### DISCUSSION TITLE
Sending to wildcard

### DISCUSSION THREAD

**Discussion by sakasaka19** [phax/phase4#201] [Q&A]
Hello Philip

In CodeList of Document Types v8.7, there are many DocumentTypes whose shceme is "peppol-doctype-wildcard".
When will we be able to send with Phase 4 for Endpoints of those DocumentType?
Right now I can't send because Lookup (pattern matching) doesn't work properly, right?

**Comment by phax** [comment]
You can already use phase4 right now - no dependency on the Document Type Identifier scheme.
What you are looking for is the `SMPClient` - the one responsible for the wildcard lookup. However, it is not done 100% automatically (from the top of my head). Ping me if you need more details - there is a 9915:helger registration with peppol-doctype-wildcard with which I tested successfully...

**Comment by sakasaka19** [comment]
Thanks for the reply.

I tried again with peppol-commons 9.1.3, 
but I could not send.

peppol-commons
peppol-smp-client/src/main/java/com/helger/smpclient/peppol/ISMPServiceMetadataProvider.java

I was wondering we use the "getWildCardServiceMetadataOrNull" method instead of the "getServiceMetadataOrNull" method to send to an Endpoint that contains WildCard.

**Reply by phax** [reply]
Ah, I think I am starting to understand. You are saying, when you use the phase4 "sending builder", it should automatically use the "wildcard resolution" instead of the "direct resolution", depending on the document type identifier scheme that is used.

I agree that is a bug and needs to be fixed asap.

**Reply by phax** [reply]
I created #203 for this matter

**Reply by phax** [reply]
It will be part of the 2.7.0 release
